### ISSUE TITLE
S3 download of files > 16MB fails

### ISSUE THREAD

**Issue by thomas-erker** [localstack/localstack#1185]
### Summary

The latest docker image version of localstack (>0.9.0) has problems when downloading files above 16MB from s3 (uploading to s3 is ok). The docker image versions 0.8.10 and 0.9.0 are not affected.

### Symptoms

Files smaller than 16MB can be downloaded without problems. Files between 16MB and around 32MB may or may not download, files above 32MB (almost) never download. The error message is:
```
download: s3://local-testing/16mb to /tmp/16mb      
download: s3://local-testing/17mb to /tmp/17mb      
download failed:  s3://local-testing/18mb to /tmp/18mb An error occurred (416) when calling the GetObject operation: Requested Range Not Satisfiable
```
### Testcase

Run the following Makefile
```
all: testfiles sync download

testfiles:
	rm -r testfiles
	mkdir -p testfiles/download
	for i in `seq 1 64`; do \
		dd if=/dev/urandom of=testfiles/$${i}mb count=$$i bs=1MB; \
	done

sync: 
	-aws --endpoint-url=http://localhost:4572 s3 mb s3://local-testing
	aws --endpoint-url=http://localhost:4572 s3 sync testfiles/ s3://local-testing/
	aws --endpoint-url=http://localhost:4572 s3 ls s3://.venv/lib/python2.7/site-packages/moto/s3/models.pylocal-testing/

download:
	for i in `seq 1 64`; do \
		aws --endpoint-url=http://localhost:4572 s3 cp s3://local-testing/$${i}mb testfiles/download/$${i}mb; \
	done

.PHONY: all testfiles sync download
```
The expected outcome is that all files can be downloaded, but instead, some files fail with the error message above.

### Analysis

The docker image of localstack after 0.9.0 uses a version of moto that contains this fix: [https://github.com/spulec/moto/pull/1999/files]
The fix replaces in-memory storage of s3 files with tempfile.SpooledTemporaryFile(max_size=DEFAULT_KEY_BUFFER_SIZE)
where DEFAULT_KEY_BUFFER_SIZE is 16MB.
Inside the localstack docker container, changing DEFAULT_KEY_BUFFER_SIZE to 32 * 1024 * 1024 in .venv/lib/python2.7/site-packages/moto/s3/models.py, removing .venv/lib/python2.7/site-packages/moto/s3/models.pyc, and killing `make infra` process (to restart moto_server) confirms that this fix is at least part of the problem. Files between 16 and 32 MB can be downloaded now, files above that size fail.

**Comment by tanishq-dubey** [comment]
I can confirm this exact behavior with a file of size 57MB.
Files of size less than 32MB work fine.

```
$> aws --endpoint-url http://localhost:4572 s3 cp s3://mybucker/folder/folder/somefile testfile 
Error for large files:
download failed: s3://mybucker/folder/folder/somefile to ./testfile An error occurred (416) when calling the GetObject operation: Requested Range Not Satisfiable

**Comment by whummer** [comment]
Thanks for the detailed bug report @thomas-erker . This should be fixed in #1200 - please re-open this issue if the problem persists. Thanks
